packages/expo-face-detector/ios/EXFaceDetector/EXFaceDetectorManager.m

```diff
@@ -187,76 +216,101 @@ - (long)_getLongOptionValueForKey:(NSString *)key
-    }];
-  }
-}
-
 - (void)_runBlockIfQueueIsPresent:(void (^)(void))block
 {
   if (_sessionQueue) {
     dispatch_async(_sessionQueue, block);
   }
 }
 
-#pragma mark - GMVDataOutputDelegate
-
-- (void)dataOutput:(GMVDataOutput *)dataOutput didFinishedDetection:(NSArray<__kindof GMVFeature *> *)results
-{
-  // Calling dataOutput:didFinishedDetection with dataOutput that in videoSettings has no information about
-  // width or height started happen after refactor: moving face detection logic from EXCameraManager to EXFaceDetectorManager.
-  // I suppose no information is provided because data output is already disconnected from the input and it has no
-  // information about the source. Let's reset the information then.
-  if (!_connected) {
-    [self _notifyOfFaces:nil];
-    return;
-  }
-  
-  AVCaptureVideoOrientation interfaceVideoOrientation = _previewLayer.connection.videoOrientation;
-  CGAffineTransform transform = [EXFaceDetectorUtils transformFromDeviceOutput:dataOutput toInterfaceVideoOrientation:interfaceVideoOrientation];
+- (void)captureOutput:(AVCaptureVideoDataOutput *)output didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer fromConnection:(AVCaptureConnection *)connection {
   
-  EXFaceEncoder *faceEncoder = [[EXFaceEncoder alloc] initWithTransform:transform];
+  dispatch_async(dispatch_get_main_queue(), ^{
+    self.interfaceOrientation = [[UIApplication sharedApplication] statusBarOrientation];
+  });
   
-  NSMutableArray<NSDictionary *> *encodedFaces = [NSMutableArray arrayWithCapacity:[results count]];
-  
-  [results enumerateObjectsUsingBlock:^(GMVFeature * _Nonnull feature, NSUInteger _idx, BOOL * _Nonnull _stop) {
-    if([feature isKindOfClass:[GMVFaceFeature class]]) {
-      GMVFaceFeature *face = (GMVFaceFeature *)feature;
-      [encodedFaces addObject:[faceEncoder encode:face]];
+  NSDate* currentTime = [NSDate new];
+  double timePassedMillis = [currentTime timeIntervalSinceDate:self.startDetect] * 1000;
+  if(timePassedMillis > self.timeIntervalMillis) {
+    
+    if(self.faceDetectionProcessing)
+    {
+      return;
     }
-  }];
-  
-  [self _notifyOfFaces:encodedFaces];
+    
+    self.startDetect = currentTime;
+    // This flag is used to drop frames when previous were not processed on time.
+    self.faceDetectionProcessing = YES;
+    
+    float outputHeight = [(NSNumber *)output.videoSettings[@"Height"] floatValue];
+    float outputWidth = [(NSNumber *)output.videoSettings[@"Width"] floatValue];
+    if(UIInterfaceOrientationIsPortrait(_interfaceOrientation)) { // We need to inverse width and height in portrait
+      outputHeight = [(NSNumber *)output.videoSettings[@"Width"] floatValue];
+      outputWidth = [(NSNumber *)output.videoSettings[@"Height"] floatValue];
+    }
+    float previewWidth =_previewLayer.bounds.size.width;
+    float previewHeight = _previewLayer.bounds.size.height;
+    
+    angleTransformer angleTransform = ^(float angle) { return -angle; };
```

Comment:
For some reason MLKit seems to have different angle orientation than GMV had. We need to invert those angles in order to stay consistent with what our JS code expects. Might be that after migration on Android I'll get rid of this transformation. Not sure though, because that would be problematic for our users, especially in older SDKs.

Reply:
I looked at that once more, and it seems to be inevitable. There are inconsistencies between Android and iOS angle orientation, and if we want to preserve one, unchanged, TS API, we need that.